### PR TITLE
Implement post-removes for DBSS objects

### PR DESCRIPTION
**Pull request by Psychotropos**
### Summary
As it stands currently, objects that go outside the scope of the DBSS for whatever technical reason (connectivity loss to the MD, process crashes et al) remain indefinitely on the state server, with no facility to inform any participants of the cluster as to what happened.

This pull request implements the "post-remove" functionality for the DBSS. Do note that as such, the recovery strategy to be used (e.g. deleting the object's representation on the state server, removing any stale references to the affected objects in the application et al) is ultimately delegated to whatever added the post-removes in question.

### Issues
My primary gripe here lies with the testing code. Some \*ahem\* creative liberty had to be taken as to how it was implemented, as the MD and the DBSS have to be executed within the scope of the test separately in order to be able to kill the DBSS and verify the post-removes are being routed as intended. Because of the aforementioned intricacy, the post-remove test has been separated from the rest of the DBSS tests. Any suggestions as to how to better integrate this, if need be, are appreciated.

### CODE REVIEW

File: src/core/msgtypes.h [hunk 1]
```diff
@@ -123,6 +123,8 @@ enum MessageTypes {
     DBSS_OBJECT_DELETE_FIELD_RAM             = 2230,
     DBSS_OBJECT_DELETE_FIELDS_RAM            = 2231,
     DBSS_OBJECT_DELETE_DISK                  = 2232,
+    DBSS_ADD_POST_REMOVE                          = 2233,
+    DBSS_CLEAR_POST_REMOVES                   = 2234,
```
Comment (mfwass) [inline src/core/msgtypes.h]: The spacing here is not consistent with the surrounding code.

File: test/common/astron.py [hunk 2]
```diff
@@ -148,6 +148,9 @@ def stop(self):
     'DBSS_OBJECT_DELETE_FIELD_DISK':             2230,
     'DBSS_OBJECT_DELETE_FIELDS_DISK':            2231,
     'DBSS_OBJECT_DELETE_DISK':                   2232,
+    'DBSS_ADD_POST_REMOVE' :                    2233,
+    'DBSS_CLEAR_POST_REMOVES':               2234,
```
Comment (mfwass) [inline test/common/astron.py]: Same here, the spacing is not similar with the surrounding code.

**Comment by Psychotropos** [comment]
Cheers for catching that, addressed as of the last commit on the branch.

MongoDB's configuration validation tests seem to fail sporadically as of current - there's nothing in it that should really cause any breakage. More importantly, this should probably not be merged into mainline quite yet - the exact details of the implementation are being re-discussed with @CFSworks.

File: src/stateserver/DBStateServer.cpp [hunk 3]
```diff
@@ -47,42 +47,76 @@ void DBStateServer::handle_datagram(DatagramHandle, DatagramIterator &dgi)
     channel_t sender = dgi.read_channel();
     uint16_t msgtype = dgi.read_uint16();
     switch(msgtype) {
-    case DBSS_OBJECT_ACTIVATE_WITH_DEFAULTS:
+    case DBSS_OBJECT_ACTIVATE_WITH_DEFAULTS: {
```
Comment (staticfox) [inline src/stateserver/DBStateServer.cpp]: Any reason for this? Kinda kills git blame.

**Comment by Psychotropos** [comment]
That styling change was in order to keep all switch cases for message types consistent, as per the new ones that required multiple lines to implement.

**Pull request closed without merging**